src/utils.ts

```diff
@@ -1,44 +1,61 @@
-import { ethers } from 'ethers'
-import { getConfig } from "./config"
+import { ethers } from "ethers";
+import { getConfig } from "./config";
+import { IWidgetThemeConfig, Orientation } from "./interfaces";
 
 const getContract = (rpcUrl: string, appAddress: string) => {
-    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
-    const appContract = new ethers.Contract(
-        appAddress, ['function walletType() view returns (uint)'],
-        provider
-    );
-
-    return appContract;
-}
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+  const appContract = new ethers.Contract(
+    appAddress,
+    ["function walletType() view returns (uint)"],
+    provider
+  );
 
+  return appContract;
+};
 
 const getWalletType = async (appId: string) => {
-    const config = getConfig();
-
-    const appAddress = await getAppAddress(appId);
-    if(!appAddress) {
-        console.log("App address not found")
-        return null
-    }
-    const c = getContract(config.RPC_URL, appAddress);
-    try {
-        const res = await c.functions.walletType();
-        const walletType = res[0].toNumber()
-        return walletType;
-    } catch(e) {
-        console.log({ e })
-        return null;
-    }
-}
+  const config = getConfig();
+
+  const appAddress = await getAppAddress(appId);
+  if (!appAddress) {
+    console.log("App address not found");
+    return null;
+  }
+  const c = getContract(config.RPC_URL, appAddress);
+  try {
+    const res = await c.functions.walletType();
+    const walletType = res[0].toNumber();
+    return walletType;
+  } catch (e) {
+    console.log({ e });
+    return null;
+  }
+};
 
 const getAppAddress = async (id: string) => {
-    const config = getConfig();
-    const res = await fetch(
-      `${config.GATEWAY_URL}/get-address/?id=` + id
-    );
-    const json = await res.json();
-    let address = json?.address;
-    return address;
-  };
-
-export { getWalletType }
\ No newline at end of file
+  const config = getConfig();
+  const res = await fetch(`${config.GATEWAY_URL}/get-address/?id=` + id);
+  const json = await res.json();
+  let address = json?.address;
+  return address;
+};
+
+const getLogo = (themeConfig: IWidgetThemeConfig, orientation: Orientation) => {
+  const { theme, assets } = themeConfig;
+  return assets.logo[theme][orientation];
+};
+
+const createDomElement = (type: string, props: object, ...children: any) => {
```

Comment:
I think we can refine the types further here. Do you mind checking if this works?

```suggestion
const createDomElement = (type: string, props: object, ...children: HTMLElement[]): HTMLElement => {
```

Reply:
@ajithranka made the suggested changes, let me know your feedback. [commit-link](https://github.com/arcana-network/wallet/pull/6/commits/f092eb90c18113394d1e3d58641ee967b4f4685a)